builtin/credential/aws/path_config_identity.go

```diff
@@ -30,27 +35,53 @@ func pathConfigIdentity(b *backend) *framework.Path {
 	}
 }
 
-func pathConfigIdentityRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	entry, err := req.Storage.Get(ctx, "config/identity")
+func identityConfigEntry(ctx context.Context, s logical.Storage) (*identityConfig, error) {
+	entryRaw, err := s.Get(ctx, "config/identity")
+	if err != nil {
+		return nil, err
+	}
+
+	var entry identityConfig
+	if entryRaw == nil {
+		entry.IAMAlias = identityAliasIAMUniqueID
+		entry.EC2Alias = identityAliasEC2InstanceID
+		return &entry, nil
+	}
+
+	err = entryRaw.DecodeJSON(&entry)
 	if err != nil {
 		return nil, err
 	}
-	if entry == nil {
-		return nil, nil
+
+	switch {
+	case entry.IAMAlias == "":
+		entry.IAMAlias = identityAliasIAMUniqueID
+	case entry.EC2Alias == "":
+		entry.EC2Alias = identityAliasEC2InstanceID
```

Comment:
I think we should change this incase both end up being "" for some reason, probably break this into two if statements